saiph/reduction/famd.py

```diff
@@ -532,3 +531,84 @@ def _compute_cos2_single_category(
         np.array(cos2) / summed_weights_without_zeros
     )
     return single_category_cos2
+
+
+def reconstruct_df_from_model(model: Model) -> pd.DataFrame:
+    """Reconstruct a DataFrame from a fitted model.
+
+    Note: if nf < df.shape[1], reconstructed df will not be exactly the same.
+
+    Parameters:
+        model: Model computed by fit.
+
+    Returns:
+        df: The reconstructed DataFrame.
+    """
+    # Step 1: Extract the necessary components from the model
+    U = model.U
+    if model.s is None:
+        raise ValueError(
+            "Model has not been fitted. Call fit() to create a Model instance."
+        )
+    S = model.s
+    Vt = model.V
```

Comment:
`Vt = V `? This does not sound intuitive...

Reply:
Yes I know. If you pay attention model.V correspond in fact to the transposed Vt. (the fit returns V=Vt)
But you are right let's simplify the comprehension